### PR TITLE
OpInfo for Slice

### PR DESCRIPTION
**Pull request by Krovatkin**
This is based on @wconstab tests from https://github.com/pytorch/pytorch/pull/84680

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -143,6 +143,45 @@ def close_to_int(x, eps=0.1):
     return (y < eps) | (y > (1 - eps))
 
 
+def sample_inputs_slice(op_info, device, dtype, requires_grad, **kwargs):
+    for i in [
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:147]: Just yield each SampleInput so this avoids stuffing them into a list and enumerating that list

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -143,6 +143,45 @@ def close_to_int(x, eps=0.1):
...
+        SampleInput(
+            input=torch.ones(3, device=device, dtype=dtype, requires_grad=requires_grad),
+            args=(0,),
+        ),
+        SampleInput(
+            input=torch.randn((20, 30, 40), device=device, dtype=dtype, requires_grad=requires_grad),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:153]: Instead of directly calling randn multiple times this should partial `make_tensor` for a more succinct and less error-prone call

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -12774,6 +12813,13 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
            supports_forward_ad=True,
            supports_fwgrad_bwgrad=True,
            supports_out=False),
+    OpInfo('slice',
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:12816]: Nice!

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:12816]: Test failures are real, though, because `torch.slice` doesn't exist. 

You can run many of the tests locally with `python test/test_ops.py -v -k slice` to see what they're doing.

I think this OpInfo needs to define its `op` metadata. This is sometimes done to wrap an operation and set a seed so it can be used for reproducible testing, like with `torch.uniform`: 

https://github.com/pytorch/pytorch/blob/90fa744c09fdbf8a2e7747ea82823714b0ee7e04/torch/testing/_internal/common_methods_invocations.py#L8164

It's also done when there's a torch method but not a torch function for an operation, like with `contiguous`: 

https://github.com/pytorch/pytorch/blob/90fa744c09fdbf8a2e7747ea82823714b0ee7e04/torch/testing/_internal/common_methods_invocations.py#L8814

And for more exotic cases, like creating a jiterated op:

https://github.com/pytorch/pytorch/blob/90fa744c09fdbf8a2e7747ea82823714b0ee7e04/torch/testing/_internal/common_methods_invocations.py#L15051

**Comment by ezyang** [comment]
I'm commandeering this PR so I can turn it into a stack

**Comment by ezyang** [comment]
commandeered at https://github.com/pytorch/pytorch/pull/85554
